client/src/components/ApplicantProfile.js

```diff
@@ -5,41 +5,61 @@ import { newProfile } from "../api/profiles";
 import Experience from "./Experience";
 
 const ApplicantProfile = () => {
+  const defaultExperience = {
+    company: "",
+    job_title: "",
+    description: "",
+    start_date: "",
+    end_date: ""
+  };
+
   const [profileData, setProfileData] = useState({
     job_title: "",
     bio: "",
     location: "",
     employment_status: "full_time"
   });
 
-  const [experienceData, setExperienceData] = useState([
-    {
-      company: "",
-      job_title: "",
-      description: "",
-      start_date: "",
-      end_date: ""
-    }
-  ]);
+  const [experiences, setExperienceData] = useState([defaultExperience]);
 
   const [profileCreated, setProfileCreated] = useState(null);
 
+  // function creates a new form with the content described below
+  const addExperience = () => {
+    setExperienceData([...experiences, defaultExperience]);
+  };
+
   const handleChange = (event, result) => {
     const { name, value } = result || event.target;
     setProfileData({ ...profileData, [name]: value });
   };
 
-  const handleExperience = ({ event, targetData, experienceIndex }) => {
+  // function merges new data with the old data
+  const updateExperiences = (value, name, index) =>
+    // experiences.map((experience, experienceIndex) => {
+    //   if (index === experienceIndex) {
+    // return { ...experience, [name]: value }; /*This return line helps to pick date from the calender*/
+    {
+      const updatedExperienceData = [...experiences];
+      updatedExperienceData[index] = {
+        ...updatedExperienceData[index],
+        [name]: value
+      };
+      return updatedExperienceData;
+    };
```

Comment:
To make this work, it has to be like this:
```
const updateExperiences = (value, name, index) => {
  const updatedExperienceData = [...experiences];
  updatedExperienceData[index] = {
    ...updatedExperienceData[index],
    [name]: value
  };
  return updatedExperienceData;
};
```

Reply:
Just checking: that's the same but without the commented code and with correct formatting, right?

Reply:
Yes